crates/ruff_linter/src/rules/pylint/rules/unnecessary_chained_comparison.rs

```diff
@@ -0,0 +1,114 @@
+
+use crate::checkers::ast::Checker;
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::{self as ast};
+
+/// ## What it does
+/// Checks for boolean operations such as `a < b and b < c`
+/// that can be refactored into a single comparison `a < b < c`.
+///
+/// ## Why is this bad?
+/// A single comparison is semantically clearer and more concise.
+///
+/// ## Example
+/// ```python
+/// a = int(input())
+/// b = int(input())
+/// c = int(input())
+/// if a < b and b < c:
+///     pass
+/// ```
+///
+/// Use instead:
+/// ```python
+/// a = int(input())
+/// b = int(input())
+/// c = int(input())
+/// if a < b < c:
+///     pass
+/// ```
+
+#[violation]
+pub struct UnnecessaryChainedComparison;
+
+impl Violation for UnnecessaryChainedComparison {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Simplified chain comparison exists between the operands.")
+    }
+}
+
+// Each integer is a unique identifier for the node.
+#[derive(Default)]
+struct Bounds {
+    lower_bound: HashSet<i32>,
+    upper_bound: HashSet<i32>,
+}
+
+fn update_bounds(
+    operator: ast::CmpOp,
+    id: String,
+    node_id: i32,
+    is_left: bool,
+    uses: &mut HashMap<String, Bounds>,
+) {
+    match operator {
+        ast::CmpOp::Lt | ast::CmpOp::LtE if is_left => {
+            uses.entry(id).or_default().lower_bound.insert(node_id);
+        }
+        ast::CmpOp::Gt | ast::CmpOp::GtE if is_left => {
+            uses.entry(id).or_default().upper_bound.insert(node_id);
+        }
+        ast::CmpOp::Lt | ast::CmpOp::LtE if !is_left => {
+            uses.entry(id).or_default().upper_bound.insert(node_id);
+        }
+        ast::CmpOp::Gt | ast::CmpOp::GtE if !is_left => {
+            uses.entry(id).or_default().lower_bound.insert(node_id);
+        }
+        _ => {}
+    }
+}
+
+fn set_lower_upper_bounds(node: &ast::ExprCompare, uses: &mut HashMap<String, Bounds>) {
+    let mut left_operand: &ast::Expr = &node.left;
+    let node_id = node as *const _ as i32;
+    for (right_operand, operator) in node.comparators.iter().zip(node.ops.iter()) {
+        if let Some(left_name_expr) = left_operand.as_name_expr() {
+            update_bounds(*operator, left_name_expr.id.clone(), node_id, true, uses);
+        }
+
+        if let Some(right_name_expr) = right_operand.as_name_expr() {
+            update_bounds(*operator, right_name_expr.id.clone(), node_id, false, uses);
+        }
+
+        left_operand = right_operand;
+    }
+}
+
+/// PLR1716
+pub(crate) fn unnecessary_chained_comparison(checker: &mut Checker, bool_op: &ast::ExprBoolOp) {
+    let ast::ExprBoolOp { op, values, range } = bool_op;
+
+    if *op != ast::BoolOp::And || values.len() < 2 {
+        return;
+    }
+
+    let mut uses: HashMap<String, Bounds> = HashMap::new();
+    for expr in values {
+        let Some(compare_expr) = expr.as_compare_expr() else {
+            return;
```

Comment:
`continue` instead of `return`
to support test case like `a < b and True and b < c`

Reply:
Fixed: https://github.com/astral-sh/ruff/pull/11807/commits/6ee4118a9fc3419069b729bff50fd9bb18059f46